### ISSUE TITLE
Cython compilation error with pyproject.toml + extensions-helpers package

### ISSUE THREAD

**Issue by aphearin** [aphearin/cython_bug_demo#2]
Cython extensions do not build properly with the package layout set up by `cookiecutter` and built using the [extensions-helpers](https://github.com/astropy/extension-helpers) package. To put this package together, I followed the instructions in the [APE 17 Migration Guide](https://docs.astropy.org/projects/package-template/en/latest/ape17.html). When I locally run `tox -e test`, I get the following traceback:

```
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/aphearin/opt/miniconda3/envs/dstar/include -fPIC -O2 -isystem /Users/aphearin/opt/miniconda3/envs/dstar/include -I/ -Ip -Ir -Ii -Iv -Ia -It -Ie -I/ -Iv -Ia -Ir -I/ -If -Io -Il -Id -Ie -Ir -Is -I/ -Iv -Ip -I/ -Ig -I9 -Iz -Iy -I6 -Iv -It -I1 -I7 -Ih -I1 -I8 -Ih -I2 -Id -In -Ip -Iq -Ix -Iz -I6 -Ij -I1 -Iw -I0 -I0 -I0 -I0 -Ig -In -I/ -IT -I/ -Ip -Ii -Ip -I- -Ib -Iu -Ii -Il -Id -I- -Ie -In -Iv -I- -Iu -Iu -I2 -In -If -Ip -I1 -Id -I/ -Io -Iv -Ie -Ir -Il -Ia -Iy -I/ -Il -Ii -Ib -I/ -Ip -Iy -It -Ih -Io -In -I3 -I. -I9 -I/ -Is -Ii -It -Ie -I- -Ip -Ia -Ic -Ik -Ia -Ig -Ie -Is -I/ -In -Iu -Im -Ip -Iy -I/ -Ic -Io -Ir -Ie -I/ -Ii -In -Ic -Il -Iu -Id -Ie -I/Users/aphearin/work/random/tmp_nohelpers/cython_bug_demo/cython_bug_demo/.tox/test/include -I/Users/aphearin/opt/miniconda3/envs/dstar/include/python3.9 -c cython_bug_demo/utils/cython_kernels/elementwise_add_cython.cpp -o build/temp.macosx-10.9-x86_64-cpython-39/cython_bug_demo/utils/cython_kernels/elementwise_add_cython.o -Ofast
      clang: error: '-I-' not supported, please use -iquote instead
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cython-bug-demo
Failed to build cython-bug-demo
ERROR: Could not build wheels for cython-bug-demo, which is required to install pyproject.toml-based projects
```

I have verified [in this alternative minimal reproducer repo here](https://github.com/aphearin/cython_pkg_demo) that the same code builds without a problem when using the old way of doing things with setup.py and without using extensions-helpers, but instead manually declaring the extensions in setup.py. Based on the success of that alternative repo, I think there is no problem with the cython, and that the problem must lie somewhere in the package layout/config. But I am new to `pyproject.toml` and also to the `extensions-helpers` package, and so hopefully this is just a simple issue resolved by a change to setup.cfg, I don't know.

Note that this is the same traceback reported [in this Halotools issue](https://github.com/astropy/halotools/pull/1047), and so it seems likely that resolving the problem in this minimal reproducer will also resolve the problem that currently blocks the release of halotools v0.8.

CC @astrofrog and/or @Cadair

**Comment by aphearin** [comment]
Still digging into this issue based on these two minimal packages. I think the config of `pyproject.toml` may actually be a red herring. In the `two_extensions` branch of [the basic working repo](https://github.com/aphearin/cython_pkg_demo), I have now adapted the code to compile two different extensions. I have tried to mirror the exact same rigamarole of the `get_extensions()` function implemented in the `extensions-helpers` package, only without the external dependency, just simply [reimplementing this function in setup.py](https://github.com/aphearin/cython_pkg_demo/blob/b500d101bd8e4bd0a252c51d0755e87554a415bb/setup.py#L25). When I try to compile the code in the two_extensions branch, I get the same error I have been getting `clang: error: '-I-' not supported, please use -iquote instead`. So I wonder whether there is some problem with the mechanism implemented by the `get_extensions` function?

**Comment by aphearin** [comment]
In [this commit](https://github.com/aphearin/cython_pkg_demo/commit/3e015d6b8a2a2a25d1ba8cb85ffa642408ee9d15) of the basic working repo, I switched from `np.get_include()` back to `['numpy']`, and now the `get_extensions()` mechanism works again.

**Comment by aphearin** [comment]
This issue is resolved by the suggestion in #3
